### ISSUE TITLE
Issue when pretrain with the snare-seq data

### ISSUE THREAD

**Issue by tang27abu** [wukevin/babel#4]
Hi Kevin,

I am very interested in the Babel model and was trying to run the model with the snare-seq data under the data folder.
I installed the environment with the conda command with the yml file. After that, I ran the command to pretrain the model but it got killed. Do you have any idea why this happened? 

```python babel/bin/train_model.py  --snareseq --outdir snareseq_model
INFO:root:PyTorch CUDA version: 10.1.243
INFO:root:Parameter data: None
INFO:root:Parameter snareseq: True
INFO:root:Parameter shareseq: None
INFO:root:Parameter linear: False
INFO:root:Parameter clustermethod: leiden
INFO:root:Parameter validcluster: 0
INFO:root:Parameter testcluster: 1
INFO:root:Parameter outdir: /storage/htc/joshilab/zl7w2/babel/snareseq_model
INFO:root:Parameter naive: False
INFO:root:Parameter hidden: [16]
INFO:root:Parameter pretrain: 
INFO:root:Parameter lossweight: [1.33]
INFO:root:Parameter optim: adam
INFO:root:Parameter lr: [0.01]
INFO:root:Parameter batchsize: [512]
INFO:root:Parameter earlystop: 25
INFO:root:Parameter seed: [182822]
INFO:root:Parameter device: 0
INFO:root:Parameter ext: pdf
INFO:root:Reading RNA data
WARNING:root:Got multiple chromosomes for gene ACEA_U3: {'chr12', 'chr19', 'chr1', 'chr11'}, skipping
WARNING:root:Got multiple chromosomes for gene SCARNA4: {'chr1', 'chr5', 'chr3', 'chr11', 'chr2'}, skipping
WARNING:root:Got multiple chromosomes for gene SCARNA15: {'chr1', 'chr5', 'chr3', 'chr11', 'chr2'}, skipping
WARNING:root:Got multiple chromosomes for gene SNORA17: {'chr16', 'chr4', 'chr2'}, skipping
WARNING:root:Got multiple chromosomes for gene SNORA50: {'chr11', 'chr4'}, skipping
WARNING:root:Got multiple chromosomes for gene Vmn1r-ps5: {'chr5', 'chr6'}, skipping
WARNING:root:Got multiple chromosomes for gene SNORA43: {'chr12', 'chr8', 'chr9'}, skipping
WARNING:root:Got multiple chromosomes for gene SNORA76: {'chrX', 'chr8'}, skipping
WARNING:root:Got multiple chromosomes for gene U3: {'chr10', 'chr11'}, skipping
INFO:root:31499 genes with known positions
Trying to set attribute `.var` of view, making a copy.
Trying to set attribute `.obs` of view, making a copy.
INFO:root:Filtering 10309 cells
INFO:root:Remaining cells after min genes: 10302
INFO:root:Remaining cells after max genes: 10302
INFO:root:Filtering 29591 vars
INFO:root:Computing size factors
INFO:root:Log transforming data
Killed
```

**Comment by wukevin** [comment]
Typically getting a `Killed` message indicates that your job is getting killed by the system job scheduler on your compute cluster. Since BABEL requires quite a bit of RAM when setting up the training data, I suspect you are running into memory limits.

If my guess is correct, try running the training script as a job where you ask your cluster's job scheduler (SGE, SLURM, etc.) for, say, 64GB of RAM and a GPU.

Let me know if this is helpful.

**Comment by tang27abu** [comment]
> Typically getting a `Killed` message indicates that your job is getting killed by the system job scheduler on your compute cluster. Since BABEL requires quite a bit of RAM when setting up the training data, I suspect you are running into memory limits.
> 
> If my guess is correct, try running the training script as a job where you ask your cluster's job scheduler (SGE, SLURM, etc.) for, say, 64GB of RAM and a GPU.
> 
> Let me know if this is helpful.

Thanks, Kevin! 
Yes, I find I didn't allocate enough memory for the model so the scanpy log1p and scale funtion got killed.
